beps/0002-dynamic-frontend-plugins/README.md

```diff
@@ -149,10 +149,57 @@ The outcome of initial testing is positive and it is possible to mix and match d
 
 The experimental code can be found in [this repository](https://github.com/scalprum/mf-mixing-experiments).
 
-**The testing so far was done only on very simple modules**. Although core React features are working (Context API and hooks), more testings needs to be done in order to declare this approach 100% reliable.
-
 So far a lot of custom code needs to be written to bridge Webpack, Rspack, @module-federation/enhanced with Vite. The first three are compatible out of the box, but Vite requires extra bridge to be able to consume/provide modules with/to other builds.
 
+#### React context and singleton sharing
+
+We can share React context and its values. Meaning a shell application (or a plugin parent) can have a context provider and a plugin will consume the context value.
+
+An example is this [package](https://github.com/scalprum/mf-mixing-experiments/tree/master/shared-package) in the experiment repo.
+
+The shell apps supply the provider and remote modules consume it. There are no issues with any combination of tooling.
+
+#### Optimized module sharing
+
+Module sharing optimizations are also working nicely. ([Optimization description](https://medium.com/@marvusm.mmi/webpack-module-federation-think-twice-before-sharing-a-dependency-18b3b0e352cb))
+
+Mixing shared scope is working between various modules using various build tools.
+
+Sample configuration:
+https://github.com/scalprum/mf-mixing-experiments/blob/master/mixed-remote-modules-collection/webpack.config.js#L27-L41
+
+```js
+const plugin = new ModuleFederationPlugin({
+  ...
+  shared: {
```

Comment:
Any suggestion on how we might do this in practice? Will we need to explicitly list every possible MUI import path?

Reply:
We are usually generating this from the node_modules. You can also write a simple tsprogram to traverse the index file and create some sort of module manifest and their paths. The manifest is particularly useful when you want to transform the import paths to ensure an optimal bundle is created.

If that is something backstage would be interested in, I can create same POC.